modules/edu.gemini.seqexec.server/src/main/scala/edu/gemini/seqexec/server/GmosControllerEpics.scala

```diff
@@ -0,0 +1,264 @@
+import scalaz.Scalaz._
+import scalaz.EitherT
+import scalaz.concurrent.Task
+
+object GmosControllerEpics extends GmosSouthController {
+  private val Log = Logger.getLogger(getClass.getName)
+
+  import EpicsCodex._
+
+  val CC = GmosEpics.instance.configCmd
+  val DC = GmosEpics.instance.configDCCmd
+
+  override def getConfig: SeqAction[GmosSouthConfig] = ???
+
+  implicit val ampReadModeEncoder: EncodeEpicsValue[AmpReadMode, String] = EncodeEpicsValue {
+    case AmpReadMode.SLOW => "SLOW"
+    case AmpReadMode.FAST => "FAST"
+  }
+
+  implicit val shutterStateEncoder: EncodeEpicsValue[ShutterState, String] = EncodeEpicsValue {
+    case OpenShutter  => "OPEN"
+    case CloseShutter => "CLOSED"
+    case _            => ""
+  }
+
+  implicit val ampGainSettingEncoder: EncodeEpicsValue[AmpGainSetting, String] = EncodeEpicsValue(_.value.toString)
+
+  implicit val ampCountEncoder: EncodeEpicsValue[AmpCount, Int] = EncodeEpicsValue {
+    case AmpCount.THREE  => 3
+    case AmpCount.SIX    => 6
+    case AmpCount.TWELVE => 12
+  }
+
+  implicit val binningEncoder: EncodeEpicsValue[Binning, Int] = EncodeEpicsValue { b => b.getValue }
+
+  implicit val disperserEncoder: EncodeEpicsValue[Disperser, String] = EncodeEpicsValue(_.sequenceValue)
+
+  implicit val disperserOrderEncoder: EncodeEpicsValue[DisperserOrder, String] = EncodeEpicsValue(_.sequenceValue)
+
+  implicit val disperserLambdaEncoder: EncodeEpicsValue[Length, String] = EncodeEpicsValue((l: Length) => l.toNanometers.toString)
+
+  implicit val beamEncoder: EncodeEpicsValue[Beam, String] = EncodeEpicsValue {
+    case OutOfBeam => "OUT-OF-BEAM"
+    case InBeam    => "IN-BEAM"
+  }
+
+  implicit val stageModeEncoder: EncodeEpicsValue[StageMode, String] = EncodeEpicsValue {
+    case StageMode.NO_FOLLOW     => "MOVE"
+    case StageMode.FOLLOW_XYZ    => "FOLLOW"
+    case StageMode.FOLLOW_XY     => "FOLLOW-XY"
+    case StageMode.FOLLOW_Z_ONLY => "FOLLOW-Z"
+  }
+
+  implicit val useElectronicOffsetEncoder: EncodeEpicsValue[UseElectronicOffset, Int] = EncodeEpicsValue(_.allow ? 1 | 0)
+
+  private def gainSetting(ampMode: AmpReadMode, ampGain: AmpGain): AmpGainSetting = (ampMode, ampGain) match {
+    case (AmpReadMode.SLOW, AmpGain.LOW)  => AmpGainSetting(2)
+    case (AmpReadMode.SLOW, AmpGain.HIGH) => AmpGainSetting(1)
+    case (AmpReadMode.FAST, AmpGain.LOW)  => AmpGainSetting(6)
+    case (AmpReadMode.FAST, AmpGain.HIGH) => AmpGainSetting(5)
+  }
+
+  private def setShutterState(s: ShutterState): SeqAction[Unit] = s match {
+    case UnsetShutter => SeqAction.void
+    case s            => DC.setShutterState(encode(s))
+  }
+
+  private def roiNumUsed(s: RegionsOfInterest): Int = s match {
+    case RegionsOfInterest(b, _) if b != BuiltinROI.CUSTOM => 1
+    case RegionsOfInterest(b, rois)                        => rois.length
+  }
+
+  case class ROI(xStart: Int, xSize: Int, yStart: Int, ySize: Int)
+
+  private def builtInROI(binning: CCDBinning, b: BuiltinROI): ROI = b match {
```

Comment:
binning is applied directly on the `roiParameters` function. but yes the parameter could be removed